app/assets/javascripts/modules/AssignedToFilter.js

```diff
@@ -0,0 +1,38 @@
+moj.Modules.AssignedToFilter = {
+
+    init: function () {
+
+        if ($('#search_query_business_unit_name_filter').length === 0 ){
+            return;
+        }
+        var module = this;
+
+        $('#search_query_business_unit_name_filter').on('keyup', function(){
+            if(this.value !== '') {
+                $('.js-all-business-units').show();
+                module.getMatchingBusinessUnits(this.value);
+            } else {
+                $('.js-all-business-units').hide();
+            }
+        }).trigger('keyup')
+    },
+
+
+    getMatchingBusinessUnits: function (searchFor){
+        $('.js-all-business-units').find('.multiple-choice')
+            .each(function( index, element ) {
+                var $elem = $(this);
+                var labelText = $elem.text().replace(/[^a-z0-9+]+/gi, '');
+                var cleanedTerm = searchFor.replace(/[^a-z0-9+]+/gi, '');
+                var check = new RegExp(cleanedTerm, 'gi')
```

Comment:
I had to talk to you to understand this a bit better ... the use a `RegExp` here implies that there could be special characters in `check` that could confuse the results, although they're being stripped out so there's no real need to use a regular expression match anyway. Not a big issue, probably worth replacing this with a substring match as and when convenient.